pkg/processor/runtime/rpc/result/result.go

```diff
@@ -31,62 +47,302 @@ type RpcLogRecord struct {
 	With     map[string]interface{} `json:"with"`
 }
 
-type Result struct {
+// singleSerialisedResult is an internal struct used for deserialising a JSON response
+// It matches the wire format and includes encoded body and metadata
+type singleSerialisedResult struct {
 	StatusCode   int                    `json:"status_code"`
 	ContentType  string                 `json:"content_type"`
-	Body         string                 `json:"body"`
-	BodyEncoding string                 `json:"body_encoding"`
 	Headers      map[string]interface{} `json:"headers"`
 	EventId      string                 `json:"event_id"`
+	Body         string                 `json:"body"`
+	BodyEncoding string                 `json:"body_encoding"`
+}
+
+// StreamStart wraps a Nuclio ResponseStream and holds the first chunk to be sent later (when reader is ready)
+type StreamStart struct {
+	*nuclio.ResponseStream
+	// firstChunk is the first chunk of data that was received
+	// we can't write it to the nuclio.ResponseStream until somebody from it, because it's blocking operation
```

Comment:
hopefully the last nitpick on this
```suggestion
	// we can't write it to the nuclio.ResponseStream until somebody reads from it, because it's blocking operation
```